src/app/services/create-hero/create-hero.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CreateHeroService {
+
+  constructor(private http: HttpClient) { }
+
+  createNewHero(heroName: string, userId) {
+    const headers = new HttpHeaders({
+      userid: userId,
+    });
+    const options = {
+      headers: headers,
+    };
+
+    return this.http.post(`http://localhost:3000/hero`, { name: heroName }, options);
```

Comment:
the url needs to be environmental

Reply:
in our environment.ts file we have environment.url which equals to that url:
export const environment = {
  production: false,
  hostname: 'http://localhost:3000',
  url: 'http://localhost:3000/hero/'
};